xesmf/frontend.py

```diff
@@ -517,19 +540,22 @@ def regrid_array(self, indata, weights, skipna=False, na_thres=1.0):
         kwargs.update(skipna=skipna, na_thres=na_thres)
 
         if isinstance(indata, dask_array_type):  # dask
-            weights = da.from_array(weights, chunks=weights.shape)
-            output_chunks = indata.chunks[:-2] + ((self.shape_out[0],), (self.shape_out[1],))
+            if output_chunks is None:
+                weights = da.from_array(
+                    self.w.data, chunks=(indata.chunksize[-2:] + indata.chunksize[-2:])
```

Comment:
> Is this default value robust ? For example, if the chunk size is larger than the output grid dimensions, does it cause a problem ?

If the chunk size is larger than the output grid dimensions, then a single chunk is used on that dimension. In the gpw example above `(5400,3000, chunks=(1000,1000))` --> `(281,297)`, the default behavior of output chunks returns outdata:`(281,297, chunks=(281,297))` because input chunks were larger than output dimensions.

> I think the code would be clearer if you wrote explicitly: ....

I agree, I'll make the change